app/v2/notifications/get_notifications.py

```diff
@@ -1,15 +1,23 @@
+import uuid
+
 from flask import jsonify, request, url_for
+from werkzeug.exceptions import abort
+
 from app import api_user
 from app.dao import notifications_dao
 from app.schema_validation import validate
 from app.v2.notifications import notification_blueprint
 from app.v2.notifications.notification_schemas import get_notifications_request
 
 
-@notification_blueprint.route("/<uuid:id>", methods=['GET'])
+@notification_blueprint.route("/<id>", methods=['GET'])
 def get_notification_by_id(id):
+    try:
+        casted_id = uuid.UUID(id)
+    except ValueError:
+        abort(404)
     notification = notifications_dao.get_notification_with_personalisation(
-        str(api_user.service_id), id, key_type=None
+        api_user.service_id, casted_id, key_type=None
```

Comment:
It might be safer to do `except Exception` as `uuid.UUID(0)` raises an AttributeError.